server/custom_vms.go

```diff
@@ -27,61 +27,195 @@ import (
 	"go.uber.org/zap"
 )
 
+type blockchainSpec struct {
+	vmName   string
+	genesis  []byte
+	subnetId *string
+}
+
 // provisions local cluster and install custom VMs if applicable
 // assumes the local cluster is already set up and healthy
-func (lc *localNetwork) installCustomVMs(ctx context.Context) error {
+func (lc *localNetwork) installCustomVMs(
+	ctx context.Context,
+	chainSpecs []blockchainSpec,
+) ([]vmInfo, error) {
 	println()
 	color.Outf("{{blue}}{{bold}}create and install custom VMs{{/}}\n")
 
 	httpRPCEp := lc.nodeInfos[lc.nodeNames[0]].Uri
 	platformCli := platformvm.NewClient(httpRPCEp)
 
-	baseWallet, avaxAssetID, testKeyAddr, err := lc.setupWallet(ctx, httpRPCEp)
+	// wallet needs txs for all previously created subnets
+	pChainCodec, err := createPChainCodec()
 	if err != nil {
-		return err
+		return nil, err
+	}
+	pTXs := make(map[ids.ID]*platformvm.Tx)
+	for _, chainSpec := range chainSpecs {
+		if chainSpec.subnetId != nil {
```

Comment:
For nil subnets, the subsecuent code create the appropiate number of subnets, and replace the nils

Reply:
Better comment: if subnet id is not nil, we need to teel the wallet about the subnet tx, so the blockchain creation does not fail. if subnet id is nil, later on we will create a subnet by using the same wallet, and it will take by its mean the tx info that needs about the created subnet

Reply:
added also comments on the code